Enriching UI docs (#3775)
temporalio/documentation: docs/develop/dotnet/enriching-ui.mdx

---
id: enriching-ui
title: Enriching the User Interface - .NET SDK
sidebar_label: Enriching the UI
description: Learn how to add contextual information to workflows and events in the Temporal UI using the .NET SDK.
toc_max_heading_level: 3
keywords:
  - ui enrichment
  - workflow context
  - static summary
  - static details
  - activity summary
  - timer summary
tags:
  - UI Enrichment
  - .NET SDK
  - Temporal SDKs
---

# Enriching the User Interface

> Enhance the Temporal UI by adding contextual information to workflows and events. This makes it easier to identify and understand workflows and their operations at a glance.

Reviewer: ```suggestion
Temporal supports adding context to workflows and events with metadata. This helps users identify and understand workflows and their operations.
```

## Adding Summary and Details to Workflows

> #### Starting a Workflow

Reviewer: ```suggestion
### Starting a Workflow
```

When starting a workflow, you can provide a static summary and details to help identify the workflow in the UI:

```csharp
using Temporalio.Client;

// Create client
var client = await TemporalClient.ConnectAsync(new("localhost:7233"));

// Start a workflow with static summary and details
var handle = await client.StartWorkflowAsync(
    (YourWorkflow wf) => wf.RunAsync("workflow input"),
    new WorkflowOptions
    {
        Id = "your-workflow-id",
        TaskQueue = "your-task-queue",
        StaticSummary = "Order processing for customer #12345",
        StaticDetails = "Processing premium order with expedited shipping"
    });
```

`StaticSummary` is a single-line description that appears in the workflow list view, limited to 200 bytes.
`StaticDetails` can be multi-line and provides more comprehensive information that appears in the workflow details view, with a larger limit of 20K bytes.

> The input format is normal Markdown excluding: images, html, and scripts.

Reviewer: ```suggestion
The input format is standard Markdown excluding images, HTML, and scripts.
```

> 

Reviewer: Although we tend to avoid screenshots, I think it would help demonstrate the usefulness of this feature by adding a screenshot of where the metadata shows up in the UI.

> You can also use `ExecuteWorkflowAsync` for synchronous execution:

Reviewer: ```suggestion
You can also use the `ExecuteWorkflowAsync` method with the same parameters:
```

```csharp
var result = await client.ExecuteWorkflowAsync(
    (YourWorkflow wf) => wf.RunAsync("workflow input"),
    new WorkflowOptions
    {
        Id = "your-workflow-id",
        TaskQueue = "your-task-queue",
        StaticSummary = "Order processing for customer #12345",
        StaticDetails = "Processing premium order with expedited shipping"
    });
```

> #### Inside the Workflow

Reviewer: ```suggestion
### Inside the Workflow
```

Within a workflow, you can get and set the _current workflow details_. Unlike static summary/details set at 

> workflow start, this value can be updated throughout the life of the workflow. Current workflow details 

Reviewer: ```suggestion
workflow start, this value can be updated throughout the life of the workflow. The `CurrentDetails` parameter
```

> also takes Markdown format (again, excluding images, html, and scripts) and can span multiple lines.

Reviewer: ```suggestion
also takes Markdown format (excluding images, HTML, and scripts) and can span multiple lines.
```

```csharp
using Temporalio.Workflows;

[Workflow]
public class YourWorkflow
{
    [WorkflowRun]
    public async Task<string> RunAsync(string input)
    {
        // Get the current details
        var currentDetails = Workflow.CurrentDetails;
        Workflow.Logger.LogInformation($"Current details: {currentDetails}");
        
        // Set/update the current details
        Workflow.CurrentDetails = "Updated workflow details with new status";
        
        return "Workflow completed";
    }
}
```

> #### Adding Summary to Activities and Timers

Reviewer: ```suggestion
## Adding Summary to Activities and Timers
```

> You can attach a `Summary` to activities when starting them from within a workflow:

Reviewer: ```suggestion
You can attach a metadata parameter `Summary` to Activities when starting them from within a Workflow:
```

```csharp
using Temporalio.Activities;
using Temporalio.Workflows;

[Workflow]
public class YourWorkflow
{
    [WorkflowRun]
    public async Task<string> RunAsync(string input)
    {
        // Execute an activity with a summary
        var result = await Workflow.ExecuteActivityAsync(
            (YourActivities act) => act.YourActivityAsync(input),
            new ActivityOptions
            {
                StartToCloseTimeout = TimeSpan.FromSeconds(10),
                Summary = "Processing user data"
            });
        
        return result;
    }
}
```

Similarly, you can attach a `Summary` to timers within a workflow:

```csharp
using Temporalio.Workflows;

[Workflow]
public class YourWorkflow
{
    [WorkflowRun]
    public async Task<string> RunAsync(string input)
    {
        // Create a timer with a summary
        await Workflow.DelayWithOptionsAsync(new DelayOptions(TimeSpan.FromMinutes(5))
        {
            Summary = "Waiting for payment confirmation"
        });
        
        return "Timer completed";
    }
}
```

> The input format for `Summary` is a string, and limited to 200 bytes.

Reviewer: Again, I think that a screenshot showing where this metadata surfaces would be useful.
